Weatherman.py

```diff
@@ -0,0 +1,277 @@
+
+
+def make_bar_chart(maximum_values, minimum_values):
+
+    display_bar_charts(maximum_values, minimum_values)
+
+    display_horizontal_bar_charts(maximum_values, minimum_values)
+
+
+def calculate_average(temperature_values):
+    total = 0
+    count = 0
+    for iterator in range(0, len(temperature_values)):
+        if temperature_values[iterator] != '':
+            total = total + float(temperature_values[iterator])
+            count += 1
+    return total / count
+
+
+def display_average_temperatures(maximum_average_temperature, minimum_average_temperature,
+                                 humidity_average):
+    print(f'Average Highest Temperature : {maximum_average_temperature}C')
+    print(f'Average Lowest Temperature : {minimum_average_temperature}C')
+    print(f'Average Humidity : {humidity_average}%')
+
+
+def display_bar_charts(maximum_temperature_values, minimum_temperature_values):
+    for iterator in range(0, len(minimum_temperature_values)):
+        if maximum_temperature_values[iterator] == '':
+            print('', end='')
+        else:
+            print(White, iterator + 1, end='')
+            display_colored_barchart(float(maximum_temperature_values[iterator]), Red)
+            print(White, f'{int(maximum_temperature_values[iterator])}C', end='')
+            print('')
+            print(White, iterator + 1, end='')
+            display_colored_barchart(float(minimum_temperature_values[iterator]), Blue)
+            print(White, f'{int(minimum_temperature_values[iterator])}C', end='')
+            print('')
+
+
+def display_horizontal_bar_charts(maximum_temperature_values, minimum_temperature_values):
+    for iterator in range(0, len(minimum_temperature_values)-1):
+        if maximum_temperature_values[iterator] == '':
+            print('', end='')
+        else:
+            print(iterator+1, end='')
+            display_colored_barchart(float(maximum_temperature_values[iterator]), Red)
+            display_colored_barchart(float(minimum_temperature_values[iterator]), Blue)
+            print(White, f'{int(maximum_temperature_values[iterator])}C-'
+                  f'{int(minimum_temperature_values[iterator])}C', end='')
+            print('')
+
+
+def display_colored_barchart(number, color):
+    number_of_stars = int(number) + 1
+    for iterator in range(1, number_of_stars):
+        print(color + '+', end='')
+
+
+# For year
+
+def make_filename(year, number):
+    month_number = str(number)
+    datetime_object = datetime.strptime(month_number, "%m")
+    month_name = datetime_object.strftime("%b")
+    file_name = f'Murree_weather_{year}_{month_name}.txt'
+    return file_name
+
+
+def run_weather_man_year(year):
+    maximum_temperatures = []
+    minimum_temperatures = []
+    minimum_temperature_dates = []
+    maximum_temperature_dates = []
+    mean_humidity = []
+    mean_humidity_dates = []                 # I need these variables in and after the for loop
+    for iterator in range(1, 13):
```

Comment:
Instead of hardcoding a number of months here, we should provide this function with files for a year, and whatever is the length of that list it should read all. In our dataset there are few cases where might not have data of all months in a year so there is no point of traversing it for fixed number.

Reply:
I have used that approach as well but ghilman bhai said that if you already know the file name then why are you making a list of files